internal/host/host.go

```diff
@@ -1412,3 +1418,89 @@ func (m *Manager) HandleReclaimBootArtifactDownload(ctx context.Context, h *mode
 func (m *Manager) HandleReclaimFailure(ctx context.Context, h *models.Host) error {
 	return m.sm.Run(TransitionTypeReclaimFailed, newStateHost(h), &TransitionArgsUnbindHost{ctx: ctx, db: m.db})
 }
+
+// autorenameHost renames a host if it is necessary and possible. For example, if the name is
+// localhost.localdomain and the host has a network interface card with MAC address
+// A8:CD:16:AE:79:01 it will be renamed to a8-cd-16-ae-79-01. Returns a boolean indicating if the
+// host was actually renamed.
+func (m *Manager) autorenameHost(host *models.Host, inventory *models.Inventory) bool {
+	if host.RequestedHostname != "" {
+		return false
+	}
+	if inventory == nil || inventory.Hostname == "" {
+		return false
+	}
+	if !funk.Contains(hostutil.ForbiddenHostnames, inventory.Hostname) {
+		return false
+	}
+	if len(inventory.Interfaces) < 1 {
```

Comment:
how can that happen? doesn't it mean that host was not able to register?

Reply:
I can imagine a scenario when the host has outside connectivity, but because we have a strictly-defined list of interface types that we collect, the list is empty.

If for example my server uses interface type `chocobomb` because I have a custom driver for it and at the same time it has no ethernet interfaces, we will end up in a scenario when the host has been discovered (because IP connectivity is available and kernel can route traffic) but Assisted does not see any interface because our collector ignores interfaces of type `chocobomb`

Reply:
good to know thanks